discovery-provider/src/tasks/karma.py

```diff
@@ -0,0 +1,64 @@
+import logging
+import time
+from src.models import Track
+from src.tasks.celery_app import celery
+from src.queries.get_trending_tracks import make_trending_cache_key, generate_unpopulated_trending
+from src.utils.redis_cache import pickle_and_set
+
+logger = logging.getLogger(__name__)
+time_ranges = ["week", "month", "year"]
+
+def get_genres(session):
+    """Returns all genres"""
+    genres = (
+        session.query(
+            Track.genre
+        ).distinct(
+            Track.genre
+        )).all()
+    genres = filter(lambda x: x[0] is not None and x[0] != "", genres)
+    return list(map(lambda x: x[0], genres))
+
+
+def update_karma(self, db, redis):
+    logger.info('Running update karma')
+    update_start = time.time()
+    with db.scoped_session() as session:
+        genres = get_genres(session)
+
+        # Make sure to cache empty genre
+        genres.append(None)
+
+        for genre in genres:
```

Comment:
can this be parallelized? not that it has to but could reduce task duration

Reply:
discussed in eng ideation